### DISCUSSION TITLE
Image Update Automation user

### DISCUSSION THREAD

**Discussion by ehudyonasi** [fluxcd/flux2#3473] [Q&A] [answered]
Hello,
I've deployed Flux with a user which has no push permissions to a certain Git Repository.
I've configured the GitRepository to use another user with the right credentials and I see it is working fine (Flux can fetch it). I've set Image Update Automation on a certain deployment and I see that it fails to push the head of the branch. I am wondering if it uses the user I've configured with the GitRepository or if it is the user I've bootstrapped Flux with.

I've searched If I can specify the username in the image update but did not find such a field.

Thanks a lot.

**Comment by kingdonb** [comment, marked as answer]
Greetings, thanks for posting! When you configure `ImageUpdateAutomation`, you set a `sourceRef` which points to a `GitRepository` there. In each `GitRepository` there is a `secretRef`. In reality, all ImageUpdateAutomation is doing with that GitRepo is to reach through the definition and access the secret.

So, ImageUpdateAutomation will use the secret of whatever (read-write) deploy key or access token is attached to the GitRepository.

As for the art-form of creating secrets in the proper form, there is `flux create secret git` as a helper which you can use to produce secrets of any of the various forms, there is an example there that uses a username and password, for example, where I think today you need a token (GitHub considers the token as the password, and the username is the username)

https://fluxcd.io/flux/cmd/flux_create_secret_git/

There are also options that show how to handle an SSH key, if you have one that you created, don't want Flux to create it in the cluster.
